### ISSUE TITLE
Problem: Unable to access local repository with Semaphore installed via Snap

### ISSUE THREAD

**Issue by kupe-dono** [ansible-semaphore/semaphore#2394]
### Issue

Hello,

I'm having trouble running a playbook from the Semaphore Web UI installed via Snap.
It seems that Snap is unable to access local folders on the host machine. Is this a known issue?

I've tried using the None, SSH and SSH Key authentication keys, and the result is the same each time -> /opt/ansible not found.

![semaphore-task-local-repo](https://github.com/user-attachments/assets/3684247d-daee-4f81-bea8-6a182a7858bb)

I've tried to use a local home dir but this time I get a permission denied:

![semaphore-task-home-repo](https://github.com/user-attachments/assets/e4862070-2e68-42e9-842c-61e646929630)

Can you please help me troubleshooting the issue?

Have a nice day/week

Kind regards,

### Impact

Ansible (task execution)

### Installation method

Snap

### Database

BoltDB

### Browser

Microsoft Edge

### Semaphore Version

The last one available from Snap for ubuntu 24

### Ansible Version

![ansible version](https://github.com/user-attachments/assets/0b2fba26-bc1a-434d-adde-2874deeb865d)

### Logs & errors

2024-10-07T10:25:58.744884+02:00 ansible semaphore.semaphored[1127]: time="2024-10-07T10:25:58+02:00" level=info msg="Task 2147483636 added to queue"
2024-10-07T10:26:00.899200+02:00 ansible semaphore.semaphored[1127]: time="2024-10-07T10:26:00+02:00" level=info msg="Set resource locker with TaskRunner 2147483636"
2024-10-07T10:26:00.899334+02:00 ansible semaphore.semaphored[1127]: time="2024-10-07T10:26:00+02:00" level=info msg="Task 2147483636 removed from queue"
2024-10-07T10:26:00.902334+02:00 ansible kernel: audit: type=1400 audit(1728289560.901:132): apparmor="DENIED" operation="capable" class="cap" profile="snap.semaphore.semaphored" pid=1127 comm="semaphore" capability=2  capname="dac_read_search"
2024-10-07T10:26:00.902348+02:00 ansible kernel: audit: type=1400 audit(1728289560.901:133): apparmor="DENIED" operation="capable" class="cap" profile="snap.semaphore.semaphored" pid=1127 comm="semaphore" capability=1  capname="dac_override"
2024-10-07T10:26:00.904191+02:00 ansible semaphore.semaphored[1127]: time="2024-10-07T10:26:00+02:00" level=info msg="Stopped running TaskRunner 2147483636"
2024-10-07T10:26:00.904250+02:00 ansible semaphore.semaphored[1127]: time="2024-10-07T10:26:00+02:00" level=info msg="Release resource locker with TaskRunner 2147483636"

![semaphore logs](https://github.com/user-attachments/assets/db698c13-68af-49e5-834b-ace5ae4c3918)

### Manual installation - system information

Linux ansible 6.8.0-45-generic #45-Ubuntu SMP PREEMPT_DYNAMIC Fri Aug 30 12:02:04 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

### Configuration

![semaphore-local-repo](https://github.com/user-attachments/assets/f19804e8-0e58-4f27-9dea-5310d3c51629)
![semaphore-local-repo-2](https://github.com/user-attachments/assets/ba6570b3-869b-43ec-a93c-f574cc2bd535)
![semaphore-playbook-config](https://github.com/user-attachments/assets/e8e9d979-1edf-4575-af76-9bfb94cf87f0)

**Comment by alyonamoskalets** [comment]
Hi!

1. I had the similar issue with SemaphoreUI [v2.10.22](https://hub.docker.com/layers/semaphoreui/semaphore/v2.10.22/images/sha256-56fbe2e86aae22fedd5dd49fb02c96544e453f7b2807a3c3df796b4e3878a622) in Docker. Everything was great, untill I wanted to start using a local repository.
2. I upgraded my SemaphoreUI with persistent data to [v2.16.45](https://hub.docker.com/layers/semaphoreui/semaphore/v2.16.45/images/sha256-9bb8e2b402320007cf0b5d963e8b1c65d866c5fa56adc8e36f82a28454ffdb47), but it has the newest Ansible installed and it does not work with my Linux servers.
3. I changed Ansible version inside the Docker container to the oldest one (v9.4.0), which has worked before, and now everything is great again.

So, to recap, the issue is in the SemaphoreUI version, probably. Consider to upgrade it, if it is applicable and possible for you.

P.S. Yes, I see, that this issue is very old, but still open. I hope, my answer would help someone who searches for a solution. :)
